packages/flutter_tools/templates/plugin_ffi/linux.tmpl/CMakeLists.txt.tmpl

```diff
@@ -0,0 +1,13 @@
+cmake_minimum_required(VERSION 3.10)
+set(PROJECT_NAME "{{projectName}}")
+project(${PROJECT_NAME} LANGUAGES CXX)
+
+# Invoke the build for native code shared with the other target platforms.
+add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/../src" "${CMAKE_CURRENT_BINARY_DIR}/shared")
+
+# List of absolute paths to libraries that should be bundled with the plugin.
+set({{projectName}}_bundled_libraries
+  # Defined in ../src/CMakeLists.txt.
+  $<TARGET_FILE:{{projectName}}>
```

Comment:
Apologies if we've discussed this and I forgot: why is this done differently from method channel plugins (where this is done in the generated file)? That would simplify both that file (since it would make the two paths the same) and this one. And it would provide one less way for people to break themselves when editing these files.

Reply:
We'd like to share the the CMakeLists.txt in src across all platforms.

We could standardise on always expecting a `CMakeLists.txt` file in `src` and standardise on a target name. However, if we have users with existing CMake builds that do not have their build file in precisely that location they cannot modify generated file that's generated every build.

Reply:
I see, it wasn't clear to me that this was to allow alternate build structures. Maybe it's worth putting a comment here and on line 11 that they could be pointed to a different build?